packages/framework/src/types/schema.types/base.schema.types.ts

```diff
@@ -14,7 +16,32 @@ export type Schema = JsonSchemaMinimal | ZodSchemaMinimal;
  */
 type InferSchema<T extends Schema, Options extends { validated: boolean }> =
   | InferJsonSchema<T, Options>
-  | InferZodSchema<T, Options>;
+  | InferClassValidatorSchema<T, Options>
+  | InferZodSchema<T, Options>
+  | never extends infer U
+  ? /*
+     * Use a distributive conditional type to detect if all inferred types are `never`.
+     * When all inferred types are `never`, return an unknown record.
+     *
+     * Each schema inferrence must return `never` type when:
+     * - The schema is generic (i.e. not a concrete schema type)
+     * - The schema is not supported (i.e. tried to specify `string` as the schema type)
+     * - The schema is undefined
+     *
+     * @see - https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types
+     */
+    [U] extends [never]
+    ? // When all inferred types are `never`, return an unknown record.
+      Record<string, unknown>
+    : // The type inferrence did not return `never`. Ensure the inferred type is a record type, as only objects are supported.
+      U extends Record<string, unknown>
+      ? // Got a record type, return it.
+        U
+      : // The schema describes a non-record type, return an error message.
+        {
+          SchemaError: `Schema must describe an object data structure. Got data type: '${Stringify<U>}'`;
+        }
+  : never;
```

Comment:
Handle all the error cases here. We now provide a nice error message when attempting to access data where an invalid schema is provided, for example:
![image](https://github.com/user-attachments/assets/9f720fbf-d182-4713-8e4e-f8c36ea4c1a9)